src/mgr/DaemonServer.cc

```diff
@@ -2545,14 +2545,38 @@ bool DaemonServer::_handle_command(
     return true;
   }
 
+  // Validate that the module is enabled
+  auto& py_handler_name = py_command.module_name;
+  PyModuleRef module = py_modules.get_module(py_handler_name);
+  ceph_assert(module);
+  if (!module->is_enabled()) {
+    ss << "Module '" << py_handler_name << "' is not enabled (required by "
+          "command '" << prefix << "'): use `ceph mgr module enable "
+          << py_handler_name << "` to enable it";
+    dout(4) << ss.str() << dendl;
+    cmdctx->reply(-EOPNOTSUPP, ss);
+    return true;
+  }
+
   // Validate that the module is active
   auto& mod_name = py_command.module_name;
+  uint64_t interval = cct->_conf->mgr_module_load_interval;
+  uint64_t timeout = cct->_conf->mgr_module_load_timeout;
+  uint64_t retries = floor(timeout / interval);
+  while (!py_modules.is_module_active(mod_name) && (retries-- > 0)) {
+    // Retry loading...
+    dout(4) << "Module '" << mod_name << "' is not active (required by command '"
+            << prefix << "'): retrying in " << interval << " secs (" << retries
+            << " retries left)." << dendl;
+    // Sleep for the retry interval
+    std::this_thread::sleep_for(std::chrono::seconds(interval));
```

Comment:
OK, so `mgr` starts taking care about delaying requests till the module gets initialized. Seems sane.

Reply:
Yeah, we basically give the mgr a little more time to retry loading the module. It's not a guarantee that the module loads in time, but if it doesn't load in the 5 seconds we give it, there is something bigger going on. But this will help account for slower hardware and any kind of latency in the cluster.

Reply:
This blocks the messenger thread. I don't think we can structure it this way. Note also 

https://github.com/ceph/ceph/pull/61325/files#diff-2de410c19e44c5a7ca3ca756e0464f8101e0218002090e36c55b3b68c40040e6R2601-R2619

which also has checks for whether the module is loaded properly. That should be consolidated.

Only the `mod_finisher` should block waiting for the module to load and there should be a condition variable that wakes the finisher thread rather than unconditionally wait `interval` seconds.